### DISCUSSION TITLE
Expansions load event. (May be changed?)

### DISCUSSION THREAD

**Discussion by zvykach** [PlaceholderAPI/PlaceholderAPI#845] [(2) Support]
Hello, I've encountered the problem that the papi expansions are loaded after the server is fully initialized. 
And if the player joins immediately after loading, the placeholders are not are not displayed correctly, since the necessary expansions have not yet been loaded.

Could you tell me why they are registered after full initialization _(most likely there is some reason)_ and whether it is possible to somehow change the moment when extensions are loaded? _(perhaps it can also be as an additional field in the config)_

Example and reason for contact
![image](https://user-images.githubusercontent.com/62516757/175065042-f6939abd-49bc-4499-8026-c1bf18856d55.png)
Messages _Done (ss.ms s)! For help, type "help" & Timings Reset_ are above

**Comment by Andre601** [comment]
There was an issue in the past where PlaceholderAPI would try to load an expansion during its enable phase, while said expansion required a plugin to work. If that plugin happens to not be enabled at this point - maybe because it depends on PlaceholderAPI - would  the loading fail.

That's why it was moved to when the server has finished loading, as that guarantees that all plugins have been enabled at least.

A fix for you could be a plugin that denies players join on the PreLoginEvent until PlaceholderAPI fired its ExpansionsLoadedEvent.
There isn't really another fix for this.

**Reply by zvykach** [reply]
Thanks. May you recommend a similar plugin?
